### PR TITLE
Feat: added elm-todomvc

### PR DESCRIPTION
**Pull request by schoening**
Ok so I ported elm-todomvc over. But there is one small problem:

image

The Json decoder fails for some reason. Can someone help me figure out why?

### CODE REVIEW

**Comment by robinheghan** [comment]
First of all, thank you for this contribution! 🥳

I'll try to take a closer look later today. Most likely something broke as part of migrating the core libraries to not use tuples or record constructors.

**Comment by schoening** [comment]
No worries @robinheghan !

I simply ran the compiler until there were no more errors, so this might not be idiomatic gren code.

Please tell me if I should change anything.

**Comment by schoening** [comment]
Ups. A little misclick there

**Comment by robinheghan** [comment]
There was a bug in `gren-lang/json` it has now been fixed.

You'll need to delete `.gren` json in your local working directory, as well as the gren cache `~/.cache/gren`. After that, it should compile and run just fine.

File: gren-todomvc/LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2014-2016, Evan Czaplicki
+All rights reserved.
```
Comment (robinheghan) [inline gren-todomvc/LICENSE:4]: Would you mind ammending this LICENSE in a similar way as in our other projects?

example: https://github.com/gren-lang/json/blob/main/LICENSE

File: gren-todomvc/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+2. Create an index.html file and copy paste the following snippet into it:
+
+```
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Elm • TodoMVC</title>
```
Comment (robinheghan) [inline gren-todomvc/README.md]: Would you mind changing all references of Elm -> Gren?

File: gren-todomvc/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <script type="text/javascript" src="gren-todomvc.js"></script>
+    <link rel="stylesheet" href="style.css" />
+  </head>
+
+  <body></body>
+
+  <script type="text/javascript">
+    console.log(window);
```
Comment (robinheghan) [inline gren-todomvc/README.md]: Remove this `console.log` maybe? 🧐

File: gren-todomvc/src/Main.gren [hunk 4]
```diff
@@ -0,0 +1,439 @@
+port module Main exposing (..)
+
+{-| TodoMVC implemented in Elm, using plain HTML and CSS for rendering.
```
Comment (robinheghan) [inline gren-todomvc/src/Main.gren]: More Elm references

File: gren-todomvc/src/Main.gren [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+        ]
+
+
+infoFooter : Html msg
+infoFooter =
+    footer [ class "info" ]
+        [ p [] [ text "Double-click to edit a todo" ]
+        , p []
+            [ text "Written by "
+            , a [ href "https://github.com/evancz" ] [ text "Evan Czaplicki" ]
```
Comment (robinheghan) [inline gren-todomvc/src/Main.gren]: Would you mind changing this to something like:

Originally written by Evan Czaplici in Elm, ported to Gren by the Gren CONTRIBUTORS

?

**Review by robinheghan — APPROVED** [review]
Thank you for porting this over!

I just have a few comments regarding some Elm => Gren renaming. Other than that, this looks good to go.

**Pull request merged**